torchgeo/datasets/pastis.py

```diff
@@ -0,0 +1,405 @@
+
+    def _verify(self) -> None:
+        """Verify the integrity of the dataset.
+
+        Raises:
+            RuntimeError: if ``download=False`` but dataset is missing or checksum fails
+        """
+        # Check if the directory already exists
+        path = os.path.join(self.root, self.directory)
+        if os.path.exists(path):
+            return
+
+        # Check if zip file already exists (if so then extract)
+        filepath = os.path.join(self.root, self.filename)
+        if os.path.exists(filepath):
+            if self.checksum and not check_integrity(filepath, self.md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            extract_archive(filepath)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                f"Dataset not found in `root={self.root}` and `download=False`, "
+                "either specify a different `root` directory or use `download=True` "
+                "to automatically download the dataset."
+            )
+
+        # Download and extract the dataset
+        self._download()
+
+    def _download(self) -> None:
+        """Download the dataset."""
+        download_url(
+            self.url,
+            self.root,
+            filename=self.filename,
+            md5=self.md5 if self.checksum else None,
+        )
+        extract_archive(os.path.join(self.root, self.filename), self.root)
+
+    def plot(
+        self,
+        sample: dict[str, Tensor],
+        show_titles: bool = True,
+        suptitle: Optional[str] = None,
+    ) -> plt.Figure:
+        """Plot a sample from the dataset.
+
+        Args:
+            sample: a sample returned by :meth:`__getitem__`
+            show_titles: flag indicating whether to show titles above each panel
+            suptitle: optional string to use as a suptitle
+
+        Returns:
+            a matplotlib Figure with the rendered sample
+        """
+        # Keep the RGB bands and convert to T x H x W x C format
+        images = sample["image"][:, [2, 1, 0], :, :].numpy().transpose(0, 2, 3, 1)
+        mask = sample["mask"].numpy()
+
+        if self.mode == "instance":
+            label = sample["label"]
+            mask = label[mask.argmax(axis=0)].numpy()
+
+        num_panels = 3
+        showing_predictions = "prediction" in sample
+        if showing_predictions:
+            predictions = sample["prediction"].numpy()
+            num_panels += 1
+            if self.mode == "instance":
+                predictions = predictions.argmax(axis=0)
+                label = sample["prediction_labels"]
+                predictions = label[predictions].numpy()
+
+        fig, axs = plt.subplots(1, num_panels, figsize=(num_panels * 4, 4))
+        axs[0].imshow(images[0] / 5000)
+        axs[1].imshow(images[1] / 5000)
+        axs[2].imshow(mask, vmin=0, vmax=19, cmap=self._cmap, interpolation="none")
```

Comment:
Wonder if we should use a different cmap for instance segmentation so that each field is a different color